packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -0,0 +1,239 @@
+      e.stopPropagation();
+
+      let backward = e.shiftKey;
+      let nextLandmark = this.getNextLandmark(e.target as HTMLElement, {backward});
+
+      // If no landmarks, return
+      if (!nextLandmark) {
+        return;
+      }
+
+      // Iterate through landmarks, starting at next landmark
+      // until we reach a landmark with a focusble element.
+      let initialNextLandmark = nextLandmark;
+      do {
+        // If something was previously focused in the next landmark, then return focus to it
+        if (nextLandmark.lastFocused) {
+          let lastFocused = nextLandmark.lastFocused;
+          if (document.body.contains(lastFocused)) {
+            lastFocused.focus();
+            return;
+          }
+        }
+
+        // Otherwise, focus the first focusable element in the next landmark
+        let leadingSentinal = nextLandmark.ref.current.previousSibling || nextLandmark.ref.current.parentElement;
+        // If we want to add a scope, this is what we're thinking:
+        // let trailingSentinal = nextLandmark.ref.current.nextSibling;
+        // let scope = [leadingSentinal, trailingSentinal];
+        let walker = getFocusableTreeWalker(nextLandmark.ref.current, {tabbable: true});
+        let nextNode = walker.nextNode() as HTMLElement;
+        if (!nextNode) {
+          walker.currentNode = leadingSentinal;
+          nextNode = walker.nextNode() as HTMLElement;
+        }
+        while (nextNode && this.closestLandmark(nextNode) !== nextLandmark) {
+          nextNode = walker.nextNode() as HTMLElement;
+        }
+        if (document.body.contains(nextNode)) {
+          nextNode.focus();
+          return;
+        }
+        
+        nextLandmark = this.getNextLandmark(nextLandmark.ref.current, {backward});
+      } while (nextLandmark !== initialNextLandmark);
+     
+    }
+  }
+
+  // Only the closest landmark cares about focus inside of it
+  public focusinHandler(e: FocusEvent) {
+    let currentLandmark = this.closestLandmark(e.target as HTMLElement);
+    this.landmarks = this.landmarks.map((landmark) => ({...landmark, ...(landmark === currentLandmark && {lastFocused: e.target as HTMLElement})}));
+  }
+}
+
+document.addEventListener('keydown', LandmarkManager.getInstance().f6Handler.bind(LandmarkManager.getInstance()), {capture: true});
+document.addEventListener('focusin', LandmarkManager.getInstance().focusinHandler.bind(LandmarkManager.getInstance()), {capture: true});
+
+/**
+ * Provides landmark navigation in an application. Call this with a role and label to register a landmark navigable with F6.
+ * @param props - Props for the landmark.
+ * @param ref - Ref to the landmark.
+ */
+export function useLandmark(props: AriaLandmarkProps, ref: MutableRefObject<HTMLElement>): LandmarkAria {
+  const {
+    role,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby
+  } = props;
+  let manager = LandmarkManager.getInstance();
+  let label = ariaLabel || ariaLabelledby;
```

Comment:
Is this the correct or at least 'best' way to register labels so we can avoid duplicate roles with the same label?

Reply:
I believe this is the best we can do right now. We can't check for what the ariaLabelledby actually resolves to. So I think it's enough to just make sure the labels aren't empty, that they are labelled by something